Algorithm.Framework/Selection/OpenInterestFutureUniverseSelectionModel.cs

```diff
@@ -0,0 +1,138 @@
+        /// <param name="futureChainSymbolSelector">Selects symbols from the provided future chain</param>
+        /// <param name="chainContractsLookupLimit">Limit on how many contracts to query for open interest</param>
+        /// <param name="resultsLimit">Limit on how many contracts will be part of the universe</param>
+        public OpenInterestFutureUniverseSelectionModel(IAlgorithm algorithm, Func<DateTime, IEnumerable<Symbol>> futureChainSymbolSelector, int chainContractsLookupLimit = 6,
+            int resultsLimit = 1) : base(TimeSpan.FromDays(1), futureChainSymbolSelector)
+        {
+            if (algorithm == null)
+            {
+                throw new ArgumentNullException(nameof(algorithm));
+            }
+
+            _algorithm = algorithm;
+            _resultsLimit = resultsLimit;
+            _chainContractsLookupLimit = chainContractsLookupLimit;
+            _exchangeHours = SecurityExchangeHours.AlwaysOpen(TimeZones.Utc);
+        }
+
+        /// <inheritdoc cref="FutureUniverseSelectionModel.Filter" />
+        protected override FutureFilterUniverse Filter(FutureFilterUniverse filter)
+        {
+            return filter.Contracts(FilterByOpenInterest(filter));
+        }
+
+        /// <summary>
+        ///     Filters a set of contracts based on open interest.
+        /// </summary>
+        /// <param name="contracts">Contracts to filter</param>
+        /// <returns>Filtered set</returns>
+        public IEnumerable<Symbol> FilterByOpenInterest(IEnumerable<Symbol> contracts)
+        {
+            return contracts.OrderBy(x => x.ID.Date).Take(_chainContractsLookupLimit).OrderByDescending(GetOpenInterest).ThenBy(x => x.ID.Date).Take(_resultsLimit);
+        }
+
+        private decimal GetOpenInterest(Symbol symbol)
+        {
+            var current = _algorithm.Time;
+            var previousBusinessDate = GetPreviousBusinessDay(_algorithm.Time, symbol);
+            var result = _algorithm.HistoryProvider.GetHistory(
+                    new[]
+                    {
+                        new HistoryRequest(
+                            previousBusinessDate,
+                            current,
+                            typeof(Tick),
+                            symbol,
+                            Resolution.Tick,
+                            _exchangeHours,
+                            TimeZones.Utc,
+                            null,
+                            true,
+                            false,
+                            DataNormalizationMode.Raw,
+                            TickType.OpenInterest
+                        )
+                    },
+                    TimeZones.Utc
+                )
+                .FirstOrDefault();
+
+            var ticks = result?.Ticks?[symbol];
+            if (ticks == null || ticks.Count == 0)
+            {
+                _algorithm.Log($"WARN: Unable to retrieve open interest for: {symbol}");
```

Comment:
Does this happen often? I think we should add "OpenInterestFutureUniverseSelectionModel" in the log message

Reply:
The contracts being provided in should all be active, so not having a value doesn't sound right to me.  If it were me, I'd rather it be an error - but I figured others would probably only want a warning.

Reply:
Note that most futures exchanges report open interest significantly after that trading session has finished, so it's quite common for it to be null (unknown) until that time.

Other exchanges provide an estimate then update it a day later.

It's certainly possible to pad it forward but it would be stale versus the other data points.

Reply:
Let me know if you're happy with the approach I've taken @RichardDale, following @Martin-Molinero's suggestions.

Reply:
There are implications for an open interest data point update changing the Universe "after the fact" especially where you have futures rolls occurring on the basis of open interest.  

Most futures traders understand this already but it's certainly a caveat emptor if you're trading spot month crude oil futures!

Otherwise, the approach looks good.

Reply:
As far as I know, there isn't any mechanism to do data revisions in Quant Connect.  So I this aspect is a bit baked into the cake, but I'll leave it up to @Martin-Molinero to decide if this can be resolved or addressed.

Reply:
> there isn't any mechanism to do data revisions in Quant Connect

Raw _backtesting_ data is available at https://www.quantconnect.com/data/tree
Regarding live trading, I believe it will drill down to the history provider being used. The history request is at the most 1 day backward with no fill forwarding from which we take the last available value. If desired, the backward history request window could be reduced, but as is seems good to me 👍